Source/WebCore/page/FrameTree.cpp

```diff
@@ -346,6 +346,22 @@ Frame* FrameTree::findBySpecifiedName(const AtomString& specifiedName, Frame& ac
     }, activeFrame);
 }
 
+Frame* FrameTree::findByPath(const Vector<size_t>& path) const
```

Comment:
This is only ever called on the main frame, and it would be problematic if this were ever called on a frame that isn't the main frame.  I think this logic would be better located on WebCore::Page.